src/etl/impl/ForwardingSource.hpp

```diff
@@ -34,12 +35,15 @@ namespace etl::impl {
 class ForwardingSource {
     util::Logger log_;
     util::requests::WsConnectionBuilder connectionBuilder_;
+    mutable std::optional<ForwardingCache> cache_;
```

Comment:
Why mutable? Since we are actually modifying the cache, should we not remove the `const` promise of the forwarding function instead?